hazelcast-jet-kafka/src/main/java/com/hazelcast/jet/connector/kafka/ReadKafkaP.java

```diff
@@ -202,35 +242,39 @@ public void init(@Nonnull Context context) {
 
         private final String topicId;
         private final Properties properties;
-        private List<Integer> ownedPartitions;
         private final Function<byte[], K> deserializeKey;
         private final Function<byte[], V> deserializeValue;
-
+        private final BiFunction<Integer, Long, Long> partitionOffsetMapper;
+        private List<Integer> ownedPartitions;
         private transient List<Processor> processors;
 
         Supplier(String topicId, Properties properties, List<Integer> ownedPartitions,
-                 Function<byte[], K> deserializeKey, Function<byte[], V> deserializeValue) {
+                 Function<byte[], K> deserializeKey, Function<byte[], V> deserializeValue,
+                 BiFunction<Integer, Long, Long> partitionOffsetMapper) {
             this.properties = properties;
             this.topicId = topicId;
             this.ownedPartitions = ownedPartitions;
             this.deserializeKey = deserializeKey;
             this.deserializeValue = deserializeValue;
+            this.partitionOffsetMapper = partitionOffsetMapper;
         }
 
-        @Override @Nonnull
+        @Override
+        @Nonnull
         public List<Processor> get(int count) {
             Map<Integer, List<Integer>> processorToPartitions =
                     IntStream.range(0, ownedPartitions.size()).boxed()
-                             .map(i -> new SimpleImmutableEntry<>(i, ownedPartitions.get(i)))
-                             .collect(groupingBy(e -> e.getKey() % count,
-                                     mapping(Entry::getValue, toList())));
+                            .map(i -> new SimpleImmutableEntry<>(i, ownedPartitions.get(i)))
```

Comment:
We align at the dot in these expressions. This is configurable in IDEA: **Preferences > Editor > Code Style > Java > Wrapping and Braces > Chained method calls > Align when multiline.**